chart/flux/templates/deployment.yaml

```
@@ -199,6 +199,7 @@ spec:
           - --ssh-keygen-dir=/var/fluxd/keygen
           - --ssh-keygen-format={{ .Values.ssh.keygen.format }}
           - --k8s-secret-name={{ .Values.git.secretName | default (printf "%s-git-deploy" (include "flux.fullname" .)) }}
+          - --k8s-secret-data-key={{ .Values.git.secretDataKey | default (printf "identity" ) }}
```

I think putting an if-statement around this instead of defaulting would be a better approach, as it prevents a double default declaration.

---

Added an if statement to check if the secretDataKey value is set.  Also changed the default value in values.yaml from "identity" to a blank string "" (although the documentation remains the same).

---

@hiddeco - Please review and resolve this conversation if you're satisfied with the change.

---

Due to the added `if`-statement, the `default` is now no longer necessary.

```suggestion
          - --k8s-secret-data-key={{ .Values.git.secretDataKey }}
```